server/grpc_service.go

```diff
@@ -908,3 +908,98 @@ func (s *Server) incompatibleVersion(tag string) *pdpb.ResponseHeader {
 		Message: msg,
 	})
 }
+
+// PrewriteMaxTS is the first part of 2PC for global MaxTS synchronization,
+// it compares the MaxTS received with each Local TSO Allocator leader's TSO
+// and prewrite it into memory to wait for the next phase of formal writing.
+func (s *Server) PrewriteMaxTS(ctx context.Context, request *pdpb.PrewriteMaxTSRequest) (*pdpb.PrewriteMaxTSResponse, error) {
+	if err := s.validateInternalRequest(request.GetHeader()); err != nil {
+		return nil, err
+	}
+	tsoAllocatorManager := s.GetTSOAllocatorManager()
+	// Get all Local TSO Allocator leaders
+	allocatorLeaders, err := tsoAllocatorManager.GetLocalAllocatorLeaders()
+	if err != nil {
+		return nil, err
+	}
+	for _, allocator := range allocatorLeaders {
+		currentLocalTSO, err := allocator.GetCurrentTSO()
+		if err != nil {
+			return nil, err
+		}
+		// Validate whether the MaxTs is greater than all other local TSOs
+		if currentLocalTSO.Physical > request.MaxTs.Physical {
+			return &pdpb.PrewriteMaxTSResponse{
+				Header:     s.header(),
+				Prewritten: false,
+				MaxLocalTs: &currentLocalTSO,
```

Comment:
It may not be the max. I think we need to iterate all allocators to find the max.

Reply:
This RPC method just implements the responding behavior of a Local TSO Allocator leader when it receives a prewriting request. After the Global TSO Allocator collects all responses, it will choose the biggest one as the max, which is not included in this PR.

Reply:
Maybe the problem is about the naming? We should not call it `MaxLocalTs` in this phase technically.

Reply:
I know. But it still needs to iterate all allocators, or the global allocator may not able to find the global max of all allocators.

Reply:
I got it.